src/core/task/tools/handlers/ReadFileToolHandler.ts

```diff
@@ -66,12 +68,40 @@ export class ReadFileToolHandler implements IFullyManagedTool {
 		}
 
 		config.taskState.consecutiveMistakeCount = 0
-		const absolutePath = resolveWorkspacePath(config.cwd, relPath!, "ReadFileToolHandler.execute")
+
+		// Resolve the absolute path based on multi-workspace configuration
+		let absolutePath: string
+		let displayPath: string = relPath!
+
+		if (config.isMultiRootEnabled && config.workspaceManager) {
+			// Parse workspace hint from the path (e.g., @frontend:src/index.ts)
+			const { workspaceHint, relPath: parsedPath } = parseWorkspaceInlinePath(relPath!)
+
+			// Create adapter for multi-workspace path resolution
+			const adapter = new WorkspacePathAdapter({
+				cwd: config.cwd,
+				isMultiRootEnabled: true,
+				workspaceManager: config.workspaceManager,
+			})
+
+			// Resolve to the correct workspace root
+			absolutePath = adapter.resolvePath(parsedPath, workspaceHint)
+
+			// Update display path for better user feedback
+			if (workspaceHint) {
+				displayPath = `@${workspaceHint}:${parsedPath}`
+			} else {
+				displayPath = parsedPath
+			}
+		} else {
```

Comment:
is it possible to handle this logic inside the current `resolveWorkspacePath` function so that most of the current code would just work?

Reply:
Done